### ISSUE TITLE
GCPCKMS auto unseal via helm_release doesn't work as intended

### ISSUE THREAD

**Issue by kenk667** [hashicorp/terraform#33766]
### Terraform Version

```shell
terraform --version
Terraform v1.5.3
on linux_amd64

Your version of Terraform is out of date! The latest version
is 1.5.6. You can update by downloading from https://www.terraform.io/downloads.html

*Apt repo doesn't appear to have 1.5.6 available

apt show versions terraform
Package: terraform
Version: 1.5.3-1
Status: install ok installed
Priority: optional
Maintainer: HashiCorp
Installed-Size: 65.1 MB
Depends: git
Homepage: https://terraform.io/
Download-Size: unknown
APT-Manual-Installed: yes
APT-Sources: /var/lib/dpkg/status
Description: Terraform enables you to safely and predictably create, change, and improve infrastructure. It is an open source tool that codifies APIs into declarative configuration files that can be shared amongst team members, treated as code, edited, reviewed, and versioned.

N: Unable to locate package versions
N: Unable to locate package versions


###

apt-cache policy terraform 
terraform:
  Installed: 1.5.3-1
  Candidate: 1.5.3-1
  Version table:
 *** 1.5.3-1 100
        100 /var/lib/dpkg/status
```

### Terraform Configuration Files

```
resource "helm_release" "vault" {
  name       = "vault"
  namespace  = "vault"
  repository = "https://helm.releases.hashicorp.com"
  chart      = "vault"
  version    = "0.20.1"

  set {
    name  = "server.ha.enabled"
    value = "true"
  }
  set {
    name  = "server.ha.raft.enabled"
    value = "true"
  }
  set {
    name  = "server.ha.raft.setNodeId"
    value = "true"
  }
  set {
    name  = "server.ha.raft.config"
    value = <<EOT
ui = false

listener "tcp" {
  tls_disable = 1
  address = "[::]:8200"
  cluster_address = "[::]:8201"
}

storage "raft" {
  path    = "/vault/data"
}

service_registration "kubernetes" {}

  seal "gcpckms" {
    cerdentials = "Service-account-key.json"
    project     = "${var.project_id}"
    region      = "${var.keyring_location}"
    key_ring    = "${var.keyring_name}"
    crypto_key  = "${var.key_name}"
  }
EOT
  }
}

```

### Debug Output

Error parsing Seal configuration: failed to encrypt with GCP CKMS - ensure the key exists and the service account has at least roles/cloudkms.cryptoKeyEncrypterDecrypter permission: rpc error: code = PermissionDenied desc = Request had insufficient authentication scopes.
2023-08-27T06:06:23.689Z [INFO]  proxy environment: http_proxy="" https_proxy="" no_proxy=""
2023-08-27T06:06:23.690Z [WARN]  storage.raft.fsm: raft FSM db file has wider permissions than needed: needed=-rw------- existing=-rw-rw----

### Expected Behavior

The auto-unseal with GCP KMS should work. If deployed via Helm with a value file that mirrors set{} stanzas it works exactly as expected where it will deploy vault and auto-unseal with supplied GCP KMS information. This is how I was able to test and narrow that it isn't a lack of permissions to the account and that the key and key ring information is accurate and works as expected. 

### Actual Behavior

When terraform applies the helm chart, no matter what variation of the helm chart version, the auto-unseal doesn't work and the deployed containers start in a crashLoop state. Digging into this problem I came across others who faced similar issues and tried applying various workarounds and none have worked. I filed a bug with Vault in case the problem is with Vault deployed via Terraform (https://github.com/hashicorp/vault/issues/22578) and now here in case the problem is with Terraform and nothing to do with Helm or Vault.

### Steps to Reproduce

*Helm provider's token is set from: data.google_client_config.current.access_token

- You will need a GKE cluster, a service account with sufficient privs (cloudkms.viewer and EncryptorDecryptor), and a KMS key and keyring.
- Either write the variables or replace the variables with strings
- Terrafrom apply then look for the running pods via kubectl get pods --namespace=vault
- If in crashloop, get logs with: kubectl logs --namespace=vault vault-0

### Additional Context

Having successfully deployed via Helm with the same credentials, key, and keyring, the error is inaccurate in that it has nothing to do with the supplied values or permissions, or the existence of keys and keyrings. What's unclear is what, if anything, Terraform may be doing before applying the Helm chart.

**Comment by apparentlymart** [comment]
Hi @kenk667! Thanks for reporting this.

I'm not familiar enough with these features to be sure, but I _think_ you're describing behavior of the `hashicorp/helm` provider rather than of Terraform Core itself. If that's true then please report this issue in [the Helm provider's repository](https://github.com/hashicorp/terraform-provider-helm) instead of here, so that the maintainers of that provider can consider it.

If you don't think that's right then please let us know and we can try to figure out how Terraform Core is involved in this behavior. Thanks!

**Comment by kenk667** [comment]
hello @apparentlymart the helm deployment with gcpckms auto-unseal works fine, but if deploying vault via helm from Terraform; the same thing fails. I tested deploying from Helm alone to see if I misconfigured something, but after discovering that the deployment worked as expected, the natural conclusion is that it's neither Helm nor me when I can only recreate the issue if deploying vault from terraform via helm_release resource. I suspect there is something happening within how the resource handles checking key status, etc, and may be dropping creds, etc. There are other similar issues to mine and I've tried every possible suggestion before I created this issue. As an aside, I also changed all of the variables to plain text and even tried using a local key file for a GCP service account with way more than enough privileges, it's the same crashLoop state and the error logs from one of the pods reporting either a missing key or keyring, or there aren't enough privs to the account. Having exhausted every permutation, here I am. 

As of now I am refactoring and restartegizing to deploy and manage Vault without terraform until either a fix or workaround is available

**Comment by kenk667** [comment]
@apparentlymart On a side note, maybe you know this answer. Why doesn't apt.releases.hashicorp.com have the latest version available? it's in the info above. The latest available is 1.5.6 but apt.releases.hashicorp.com only has 1.5.3 available. Having gone from back in the day of managing evolving terraform installation methods, I really don't want to wget a package. If 1.5.6 has a fix for myself, I certainly will, but I'd expect 1.5.6 to be available to install per the Hashicorp documented installation method (I'm on Ubuntu).

**Comment by crw** [comment]
> Why doesn't apt.releases.hashicorp.com have the latest version available?

If your distribution is supported (https://www.hashicorp.com/official-packaging-guide?ajs_aid=2f154dd0-0d8a-4e2c-86f4-fd81f02404cb&product_intent=terraform) then please open a new issue with as much information as possible. I could not reproduce this on ubuntu lunar:

```
Get:1 https://apt.releases.hashicorp.com lunar/main amd64 terraform amd64 1.5.6-1 [21.8 MB]
```

**Comment by kenk667** [comment]
lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
Codename:	jammy

**Comment by kenk667** [comment]
I tend to stick with LTS releases and would expect support for LTS versions, @crw can you try reproducing the error with 22.04?

**Comment by crw** [comment]
```
Get:1 https://apt.releases.hashicorp.com jammy/main amd64 terraform amd64 1.5.6-1 [21.8 MB]
```
If you want to continue this discussion, please open a new issue. Thanks!

**Comment by kenk667** [comment]
@crw --> https://github.com/hashicorp/terraform/issues/33782

Am interested in discovering the original bug reported for why the combination of TF 1.5.3 deploying vault through helm_release resource and using set{} to pass GCPCKMS values fails

**Comment by crw** [comment]
@kenk667 If you haven't already, please consider opening an issue in the [the Helm provider's repository](https://github.com/hashicorp/terraform-provider-helm) -- this is likely the source of the issue, however we will leave this issue open since it did not seem to be resolved whether it could be a core issue. Thanks!

**Comment by kenk667** [comment]
@crw see this: https://github.com/hashicorp/vault/issues/22578#issuecomment-1706817089

I have updated to the latest version of TF (1.5.6) and tried potential fixes from a similar issue I filed with vault. The behavior for how gcpckms auto-unseal works seems to have some other impact by terraform. This is evident when all fixes have been applied from the vault side, and I can deploy without issue if deployed directly from Helm. This means to me that terraform is doing something to change how creds are passed between terraform and either helm and/or vault
